flloat/ltlf.py

```diff
@@ -479,6 +531,23 @@ def _delta(self, i: PropositionalInterpretation, epsilon=False):
             ]
         )
 
+    def to_ldlf(self):
+        """Convert the formula to LDLf."""
+        f1 = LDLfNot(self.formulas[0].to_ldlf())
+        f2 = (
+            LTLfRelease(self.formulas[1:]).to_ldlf()
+            if len(self.formulas) > 2
+            else LDLfNot(self.formulas[1].to_ldlf())
+        )
+        return LDLfNot(
+            LDLfDiamond(
+                RegExpStar(
+                    RegExpSequence([RegExpTest(f1), RegExpPropositional(PLTrue())])
+                ),
+                LDLfAnd([f2, LDLfNot(LDLfEnd())]),
+            )
```

Comment:
```
!(!f1 Until !f2) = f1 Release f2

<(!f1?;true)*>(!f2 & !end) = [(f1?;true)*](f2 | end)

```

Reply:
Also change the Not at 540

Reply:
I think it should be like this:
```
!<(!f1?;true)*>(!f2 & !end) = [((!f1)?;true)*](f2 | end)
```

Reply:
no, I believe the not must be forwarded to `f1` in the test regex

Reply:
resolved privately, the regex should not be touched by the duality of diamond and box operators.